lesson03/gulpfile.js

```diff
@@ -0,0 +1,135 @@
+var destDir = 'bin';
+var gulp = require('gulp');
+var autoprefixer = require('gulp-autoprefixer');
+var bower = require('gulp-bower');
+var bowerMain = require('bower-main');
+var gulpif = require('gulp-if');
+var concat = require('gulp-concat');
+var connect = require('gulp-connect');
+var less = require('gulp-less');
+var argv = require('yargs').argv;
+var debug = require( 'gulp-debug' );
+var htmlhint = require('gulp-htmlhint');
+var clean = require( 'gulp-clean' );
+var livereload = require('gulp-livereload');
+var csscomb = require('gulp-csscomb');
+var jscs = require('gulp-jscs');
+var jshint = require('gulp-jshint');
+var runSequence = require('run-sequence');
+var sourcemaps = require('gulp-sourcemaps');
+var cssnano = require('gulp-cssnano');
+var minifyCss = require('gulp-minify-css');
+var uglify = require("gulp-uglify");
+
+gulp.task('default', ['libs', 'build']);
+
+gulp.task('defaultAdd', ['css', 'js', 'html', 'images'])
+
+gulp.task('copy-static', function () {
+    return gulp.src(['images/**/*.{png,jpg,svg}', '*.html', '**.*.js'])
+        .pipe( gulp.dest(destDir) );
+});
+
+gulp.task('bower', function () {
+    return bower('libs');
+});
+
+gulp.task('build', ['copy-static', 'css']);
+
+gulp.task('libs',['bower'], function () {
+    return gulp.src( bowerMain( '*.js', '*.min.js' ).minified, { base: destDir })
+        .pipe(gulp.dest(destDir+'/libs/'));
+});
+
+gulp.task('images', function () {
+    return gulp.src(['**/*.@(png|jpg|svg)', '!**/node_modules/**', '!**/libs/**', '!**/bin/**']) //Если bin не исключить, то начинает копировать сам себя
```

Comment:
You can make `'!**/node_modules/**', '!**/libs/**', '!**/bin/**` as a var. But this is up to you :)
```js
var defaultExclude = ['!**/node_modules/**', '!**/libs/**', '!**/bin/**'];

gulp.src(['**/*.@(png|jpg|svg)'].concat(defaultExclude))
gulp.src(['**/*.html'].concat(defaultExclude))

```